nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/blob/ColdStorageHelper.java

```diff
@@ -75,6 +83,61 @@ public static DocumentModel moveContentToColdStorage(CoreSession session, Docume
         return session.saveDocument(documentModel);
     }
 
+    /**
+     * Requests a retrieval of the cold storage content associated with the document of the given {@link DocumentRef}.
+     *
+     * @param session the core session
+     * @param documentRef the document reference
+     * @param numberOfDaysOfAvailability number of days that you want your cold storage content to be accessible after
+     *            restoring
+     * @apiNote This method will initiate a restoration request, calling the {@link Blob#getStream()} during this
+     *          process doesn't mean you will get the blob's content.
+     * @return the updated document model if the retrieve succeeds
+     * @throws NuxeoException if there is no cold storage content associated with the given document, or if it is being
+     *             retrieved
+     */
+    public static DocumentModel requestRetrievalFromColdStorage(CoreSession session, DocumentRef documentRef,
+            int numberOfDaysOfAvailability) {
+        DocumentModel documentModel = session.getDocument(documentRef);
+        log.debug("Retrieve from cold storage the content of document: {} for: {} days", documentModel,
+                numberOfDaysOfAvailability);
+
+        if (!documentModel.hasFacet(FacetNames.COLD_STORAGE)
+                || documentModel.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY) == null) {
+            throw new NuxeoException(String.format("No cold storage content defined for document: %s.", documentModel),
+                    SC_NOT_FOUND);
+        }
+
+        Serializable beingRetrieved = documentModel.getPropertyValue(COLD_STORAGE_BEING_RETRIEVED_PROPERTY);
+        if (Boolean.TRUE.equals(beingRetrieved)) {
+            throw new NuxeoException(
+                    String.format("The cold storage content associated with the document: %s is being retrieved.",
+                            documentModel),
+                    SC_FORBIDDEN);
+        }
+
+        try {
+            Blob coldContent = (Blob) documentModel.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY);
+            String coldContentKey = getContentBlobKey(coldContent);
+            Duration duration = Duration.of(numberOfDaysOfAvailability, ChronoUnit.DAYS);
+            BlobUpdateContext updateContext = new BlobUpdateContext(coldContentKey).withRestoreForDuration(duration);
+            Framework.getService(BlobManager.class).getBlobProvider(coldContent).updateBlob(updateContext);
```

Comment:
There's something we probably have to refactor a bit to make this easier on the caller here. Later.

Reply:
You mean later / other ticket(s) ?

Reply:
Yes we'll do it later, that was just FYI.